app/views/layouts/application.html.erb

```diff
@@ -23,6 +23,45 @@
       <%= render_phase_label @content_item, content_for(:phase_message) %>
       <%= yield %>
     </main>
+
+    <div class="improve-this-page" data-module="improve-this-page">
+      <div class="js-prompt">
+        <span class="improve-this-page__is-useful-question">Is this page useful?</span>
+        <%= link_to "Yes", contact_govuk_path, class: 'js-page-is-useful' %>
+        <%= link_to "No", contact_govuk_path, class: 'js-offer-feedback' %>
+        <div class="improve-this-page__anything-wrong">
+          <%= link_to "Is there anything wrong with this page?", contact_govuk_path, class: 'js-offer-feedback' %>
+        </div>
+      </div>
+      <div class="js-feedback-form js-hidden">
+        <div class="grid-row">
+          <div class="column-two-thirds">
+            <div class="js-errors"></div>
+            <form>
+              <input type="hidden" name="path" value="<%= request.original_url -%>">
+              <div class="improve-this-page__section">
+                <label>How should we improve this page?</label>
+                <textarea name="description" rows="5"></textarea>
```

Comment:
I've added commits to bind to the submit event on the form and have added the required attribute. I'm going to leave the code in which renders the errors inline. It's only 3-4 lines and a test which doesn't seem like much of a cost for not having to re-add it if we add other validations to the support-api.